src/ui/public/visualize/loader/embedded_visualize_handler.js

```diff
@@ -20,6 +20,32 @@ export class EmbeddedVisualizeHandler {
     });
   }
 
+  /**
+   * Update properties of the embedded visualization. This method does not allow
+   * updating all initial parameters, but only a subset of the ones allowed
+   * in {@link VisualizeLoaderParams}.
+   *
+   * @param {Object} [params={}] The parameters that should be updated.
+   * @property {Object} [timeRange] A new time range for this visualization.
+   * @property {Object} [dataAttrs] An object of data attributes to modify. The
+   *    key will be the name of the data attribute and the value the value that
+   *    attribute will get. Use null to remove a specific data attribute from the visualization.
+   */
+  update(params = {}) {
+    this._scope.$evalAsync(() => {
+      if (params.timeRange) {
```

Comment:
This basically prevents nulling / undefining the timeRange. Just want to double-check that that is what we want. If we want to allow nulling the timeRange, this should be `if (params.hasOwnProperty('timeRange'))` or something.

Reply:
That's indeed desired behavior. We are requiring specifying a timeRange. After this has merged, we will basically remove all functionality in visualize to read out timeRange from the global state and always rely on the explicitly passed one, so we don't ever want this to be `null`.

Reply:
ehm .... i think not setting the time range still makes sense (or nulling it), since not all indexes are time based this parameter might be obsolete in some cases. We shouldn't force users to use this, i can completely see the use case where somebody creates a plugin for his specific usecase where he has no time based indexes.

Reply:
I totally agree, you might have visualizations where you don't need the timeRange. So we cannot check if it's set initially, but I still think we could forbid nulling it, since I don't see a reason why you would want to have it initially, but then remove it. In the cases where the index would not need a timeRange you would not set a timeRange initially and thus never need to null it?

Reply:
any specific reason why you want to prevent nulling it ? (i see it as an unnecessary restriction, which we are not sure at this point if its useful or not)